### ISSUE TITLE
Crash in wx.DropTarget.OnData when retrieving second CustomDataObject from DataObjectComposite

### ISSUE THREAD

**Issue by vScourge** [wxWidgets/Phoenix#550]
Windows 8.1 Enterprise
wxPython version 4.0.0a3
custom build
Python version 3.5.2
custom build

Using drag and drop with a DataObjectComposite object.  Inside DropTarget.OnData, wxPython will crash when the second CustomDataObject is retrieved by calling GetData().  The first one always appears to work fine.

I'm attaching a modified version of the CustomDragAndDrop.py demo file that illustrates the problem.  The crash happens at line 172.
[CustomDragAndDrop.zip](https://github.com/wxWidgets/Phoenix/files/1322150/CustomDragAndDrop.zip)

**Comment by HelioGuilherme66** [comment]
Maybe you should try with the newest wxPython version 4.0.0b2.
I tried your example, there is no crash but the objects are not dragged.
This is the stdout:
```
C:\tmp\CustomDragAndDrop>python CustomDragAndDrop.py
Begining DragDrop
DragDrop completed: 1
Begining DragDrop
OnEnter: 107, 2, 3
OnDrop: 109 88
OnData: 109, 88, 3
Traceback (most recent call last):
  File "CustomDragAndDrop.py", line 169, in OnData
    lines = cPickle.loads(linesdata)
TypeError: loads() argument 1 must be string, not memoryview
DragDrop completed: 1
Begining DragDrop
DragDrop completed: 1
Begining DragDrop
OnEnter: 129, -1, 3
OnDrop: 132 24
OnData: 132, 24, 3
Traceback (most recent call last):
  File "CustomDragAndDrop.py", line 169, in OnData
    lines = cPickle.loads(linesdata)
TypeError: loads() argument 1 must be string, not memoryview
DragDrop completed: 1
Begining DragDrop
OnEnter: 24, 0, 3
OnDrop: 51 81
OnData: 51, 81, 3
Traceback (most recent call last):
  File "CustomDragAndDrop.py", line 169, in OnData
    lines = cPickle.loads(linesdata)
TypeError: loads() argument 1 must be string, not memoryview
DragDrop completed: 1
Begining DragDrop
OnEnter: 163, -1, 3
OnDrop: 130 69
OnData: 130, 69, 3
Traceback (most recent call last):
  File "CustomDragAndDrop.py", line 169, in OnData
    lines = cPickle.loads(linesdata)
TypeError: loads() argument 1 must be string, not memoryview
DragDrop completed: 1
Begining DragDrop
OnEnter: 77, -2, 3
OnDrop: 73 68
OnData: 73, 68, 3
Traceback (most recent call last):
  File "CustomDragAndDrop.py", line 169, in OnData
    lines = cPickle.loads(linesdata)
TypeError: loads() argument 1 must be string, not memoryview
DragDrop completed: 1
```

**Comment by vScourge** [comment]
Interesting, thanks.  I'll give that a try.

**Comment by RobinD42** [comment]
It's crashing for me in my current workspace so a newer version likely didn't help. @HelioGuilherme66 probably just got lucky :wink: I'll see if I can spot the problem...

**Comment by RobinD42** [comment]
Oops, I didn't look at the tracebacks in @HelioGuilherme66's message, it's a different problem which is happening before the crash would have.  Were you using Python 2.7?

**Comment by RobinD42** [comment]
Okay, I've fixed the crash, but discovered something new along the way.  Only the data object with the first matching format is transferred for composite data objects when you call `self.GetData`. I don't see an easy way to get more than one matching format from the source, so you'll probably need to derive a custom data object type from wx.DataObject if you need to do that.

**Comment by RobinD42** [comment]
(I also fixed the demo sample for Python 2.7)
